pkg/provisioner/ironic/ironic.go

```diff
@@ -231,9 +247,21 @@ func New(host metal3v1alpha1.BareMetalHost, bmcCreds bmc.Credentials, publisher
 		if err != nil {
 			return nil, err
 		}
+
+	}
+	// TODO(iurygregory): Update here if we want to consider a trusted CAFile for the bmc.
+	if bmcClientSingleton == nil {
+		tlsConf := clients.TLSConfig{
+			TrustedCAFile:      "",
+			InsecureSkipVerify: host.Spec.BMC.DisableCertificateVerification,
+		}
+		bmcClientSingleton, err = clients.BMCClient(tlsConf)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
@andfasano since the `ironicProvisioner`  should be per host, i.e. per ironic node, this would make sense I would say.

Reply:
A new provisioner gets created for every reconcile loop, and in the current proposal a singleton client would be shared for all the different hosts

Reply:
I see, so we would be trying to reuse the `http.Client` in different BMC's =(

Reply:
Can we maybe go down a different path? Make sure we (successfully) validate each host only once? After all, we only need a best effort check anyway/

Reply:
There's nothing wrong with using a single Client for everything, just as there's nothing wrong with using a single web browser to visit multiple websites. It's recommended [in the documentation](https://golang.org/pkg/net/http/#Client) even:

> Clients should be reused instead of created as needed.

In fact, I'd consider it compulsory because otherwise things get very leaky: see #709

Reply:
Thanks everyone! So I will keep this approach.